src/createServer.js

```diff
@@ -1,11 +1,280 @@
   // Use express to create a server
   // Add a routes to the server
   // Return the server (express app)
+
+  const app = express();
+
+  const users = [];
+
+  app.get('/users', express.json(), (req, res) => {
+    if (!users) {
+      res.statusCode = 200;
+
+      res.send([]);
+
+      return;
+    }
+
+    res.statusCode = 200;
+
+    res.send(users);
+  });
+
+  app.post('/users', express.json(), (req, res) => {
+    const { name } = req.body;
+
+    if (!name) {
+      res.statusCode = 400;
+
+      res.send();
+
+      return;
+    }
+
+    const user = {
+      name,
+      id: users.length + 1,
+    };
+
+    users.push(user);
+
+    res.statusCode = 201;
+
+    res.send(user);
+  });
+
+  app.get('/users/:id', express.json(), (req, res) => {
+    const { id } = req.params;
+
+    const findUser = users.find((user) => user.id === Number(id));
+
+    if (!findUser) {
+      res.statusCode = 404;
+
+      res.send();
+
+      return;
+    }
+
+    if (Number.isNaN(Number(id))) {
+      res.statusCode = 400;
+
+      res.send();
+
+      return;
+    }
+
+    res.statusCode = 200;
+
+    res.send(findUser);
+  });
+
+  app.delete('/users/:id', express.json(), (req, res) => {
+    const { id } = req.params;
+
+    const findUser = users.find((user) => user.id === Number(id));
+
+    if (!findUser) {
+      res.statusCode = 404;
+
+      res.send();
+
+      return;
+    }
+
+    if (Number.isNaN(Number(id))) {
+      res.statusCode = 400;
+
+      res.send();
+
+      return;
+    }
+
+    users.splice(users.indexOf(findUser), 1);
+
+    res.statusCode = 204;
+
+    res.send();
+  });
+
+  app.patch('/users/:id', express.json(), (req, res) => {
+    const { id } = req.params;
+
+    const findUser = users.find((user) => user.id === Number(id));
+
+    if (!findUser) {
+      res.statusCode = 404;
+
+      res.send();
+
+      return;
+    }
+
+    if (Number.isNaN(Number(id))) {
+      res.statusCode = 400;
+
+      res.send();
+
+      return;
+    }
+
+    const { name } = req.body;
+
+    if (!name) {
+      res.statusCode = 400;
+
+      res.send();
+
+      return;
+    }
+
+    findUser.name = name;
+
+    res.statusCode = 200;
+
+    res.send(findUser);
+  });
+
+  const expenses = [];
+
+  app.get('/expenses', express.json(), (req, res) => {
+    const { userId, category, from, to } = req.query;
+
+    if (category) {
+      const filteredExpenses = expenses.filter(
+        (expense) => expense.category === category
+      );
+
+      res.statusCode = 200;
+
+      res.send(filteredExpenses);
+
+      return;
+    }
+
+    if (userId) {
+      const filteredExpenses = expenses.filter(
+        (expense) => expense.userId === Number(userId)
+      );
+
+      res.statusCode = 200;
+
+      res.send(filteredExpenses);
+
+      return;
+    }
+
+    if (from && to) {
+      const filteredExpenses = expenses.filter(
+        (expense) => expense.spentAt >= from && expense.spentAt <= to
+      );
+
+      res.statusCode = 200;
```

Comment:
Maybe code would be more elegant in you would split this logic into 3 different methods. What do you think?